File: 2015.1.lua
-- AoC 2015 Day 1 --

local input = assert(io.open"2015.1.in"):read"*a"

local p1, p2 = 0, nil
for i, char in input:gmatch"()(.)" do
	if char == "(" then p1 = p1 + 1 end
	if char == ")" then p1 = p1 - 1 end
	if p1 < 0 and not p2 then
		p2 = i
	end
end

print("A: "..p1)
print("B: "..p2)

File: 2015.1.py
# AoC 2015 Day 1 #

with open("2015.1.in", "r") as file:
    data = file.read()

p1 = sum(1 if char == "(" else -1 for char in data)

p = 1
p2 = [p := p + (1 if char == "(" else -1) for char in data].index(-1)

print(f"A: {p1}")
print(f"B: {p2}")
